### PR TITLE
Fix failing test

### PR DESCRIPTION
**Pull request by fingolfin**
The string returned by DrawSplash() can differ based on the order of
some dict entries, and thus fails for me. Resolve this by instead only
performing a cursory check of the return value.

### CODE REVIEW

**Comment by fingolfin** [comment]
@mcmartins would you be willing to just merge this and make a new release, with otherwise no changes? It shouldn't be worse than the previous release this way, but at least it will pass the CI tests in the Package Distribution, which would be quite helpful, to establish a baseline of "all packages pass their tests"

**Comment by mcmartins** [comment]
Hi @fingolfin , give me some days to setup all the environment and make sure the project can be released again, as things changed quite a bit since last release.

**Comment by fingolfin** [comment]
@mcmartins actually, there was a single change since the last release, namely https://github.com/gap-packages/francy/commit/3d33f91089aab7dba9826cdbc7ad1d248acc5205 -- my suggestion was to add one more (namely this PR), then make a new release. Granted, this won't fix any of the other issues, but it should be 100% safe in the sense that it will be no worse than what it is out there. It also has the advantage of being easy (i.e. I could do it, right now, in under 5 minutes -- in fact, if you tell me you are OK with this plan, I'd be happy to do it ASAP).

Of course a more comprehensive update would be nice, but I am afraid how much work that'll take, and thus how long -- if it means waiting another half year, that wouldn't be nice (no offense intended -- I know how it is, and I appreciate your willingness to look into this, but I also know how easy it is for these kind of things to spiral out of control, and I am sure you have many other things to do, most probably more important to you, too!)

**Comment by mcmartins** [comment]
@fingolfin when I say “things changed quite a bit” I mean with Travis CI / NPM / etc, and the pipeline I had set up, and I don’t believe the project release will just work as it is, even if the only change is this test case. Indeed, I do have other important commitments, and I’ll try my best to sort this out…

**Comment by fingolfin** [comment]
tl;dr: it'd be great if you let me release 1.2.5 with this fix, in a hackish-but-safe manner sketched below, with zero work required by you.

long version:

My idea for a new francy release was to use a more modest approach that in this case however should be fully sufficient:
1. merge this PR2. 
1. add one more commit to master to set the version to 1.2.5, update the release date
1. manually tag `master` as `v1.2.5`
1. download the 1.2.4 tarball
1. apply the `git diff v1.2.4..v1.2.5` to the content of that directory
1. run `makedoc.g` to regenerate all derived files
1. (optional but recommended) remove "garbage" files created by LaTeX in doc dir
1. tar everything back together into a 1.2.5 tarball
1. manually upload the new tarball to GitHub
1. manually update the `gh-pages` branch for the new release

I just experimentally did steps 1.-8. (of course with 1.-3. only locally!) and it seemed to work great.

Of course this is just a stop-gap measure, and a *real* new release would be great, and yet muuuuch more work. But this stop-gap measure would be great help for us on the GAP side, as we could finally get rid of a constantly failing `francy` test suite in a clean way (I mean, I could hack around it by patching the `francy` test suite or something like that, but there are several places these tests run, and it'd quickly become a mess.


Now, I have zero expectations that you do this, after all, there is nothing in there for you (I hope this gets across right; I don't mean this negative, it's just a fact: this is about scratching an itch of the GAP team, not one of you as a package author!).

Hence, I'd be happy to perform the steps sketched above. I have all the access permissions necessary. But I do not want to do this without your consent. (And if for some reason you are radically opposed, well, I'll have to look into hackish workarounds elsewhere...)

**Comment by mcmartins** [comment]
@fingolfin , no objection at all, I'm happy with your plan. 

Furthermore, I'll make sure I change things to github actions and that the JS part can be released next time without my involvement (I'm looking into prebuilt lab extensions as well to ease the installation).

**Comment by fingolfin** [comment]
Excellent! The easier it is to release, the better :-)

**Comment by fingolfin** [comment]
Actually, there might be an issue with my plan, but @mcmartins might know better: there is one place where francy uses its own version:
```gap
InstallMethod(Draw,
  "a canvas",
  true,
  [IsCanvas],
  0,
function(canvas)
  local object;
  object := rec();
  object!.mime    := FrancyMIMEType;
  object!.version := InstalledPackageVersion("francy");
  object!.canvas  := Sanitize(canvas);
  return Objectify(
    JupyterRenderableType, 
    rec(
      data := rec((FrancyMIMEType) := GapToJsonString(object)),
      metadata := rec((FrancyMIMEType) := rec())
    )
  );
end);
```
Not sure if that will work if the npm packages stay at version 1.2.4? If not, then a workaround would be to hardcode "1.2.4" above (marking the line with a big FIXME comment, and I'd also open a PR to revert that change?).

Thoughts?

**Comment by mcmartins** [comment]
There might be a warning in the browser console if the versions don't match so I agree with hardcoding it to 1.2.4 with a FIXME comment. I'll address all these on the changes I'm working to make the package more "standard" to gap. 

Thanks for looking into this @fingolfin !

**Comment by fingolfin** [comment]
I've now executed the plan. PR #78 reverts my hack. I suggest to not merge it yet in case a 1.2.6 release is necessary to fix up any problems with this hack-release may cause (and perhaps it will never be merged if you just end up rewriting that code, who knows -- but at least it'll server as a reminder that the hack is in there).
